src/hangman.rs

```diff
@@ -0,0 +1,106 @@
+#[derive(Default)]
+pub struct Hangman {
+    word: String,
+    attempts: usize,
+    misses: [char; 5],
+    hits: Vec<char>,
+}
+
+pub enum AttemptResult {
+    CanKeepPlaying,
+    Win,
+    Hanged,
+}
+
+const SYMBOL_LETTER_TO_COMPLETE:char = '_';
+
+impl Hangman {
+    pub fn new(word: String) -> Hangman {
+        let vec_size = word.len();
+        Hangman { 
+            word, 
+            attempts: 0, 
+            misses: Default::default(), 
+            hits: vec![SYMBOL_LETTER_TO_COMPLETE; vec_size],
+        }
+    }
+
+    fn guess_letter(&mut self, l:char) -> AttemptResult {
+        
+        if self.hits.contains(&l) || self.misses.contains(&l) {
+            return AttemptResult::CanKeepPlaying;
+        }
+
+        let mut guessed_letter = false;
+        let mut missing_letters = false;
+
+        for (i, c) in self.word.chars().enumerate(){
+            if c == l {
+                self.hits[i] = l;
+                guessed_letter = true;
+            }
+
+            if self.hits[i] == SYMBOL_LETTER_TO_COMPLETE {
+                missing_letters = true;
+            }
+        }
+
+        if !missing_letters {
+            AttemptResult::Win
+        } else if guessed_letter {
+            AttemptResult::CanKeepPlaying
+        } else {
+            self.misses[self.attempts] = l;
+            self.attempts += 1;
+
+            if self.attempts == 5 {
+                AttemptResult::Hanged
+            } else {
+                AttemptResult::CanKeepPlaying
+            }
+        }
+    }
+
+    pub fn show_game(&self) {
+        const TITLE:&str  = "Bienvenido al hangman de LAMBDA!";
+        const WORD:&str = "La word hasta el momento es:";
+        const ATTEMPTS:(&str, &str) = ("Te quedan", "attempts.");
+        const LETTERS:&str = "Letras usadas:";
+        const PROMPT:&str = "Ingresa una letra:";
+
+
+        let mut guessed_vec:Vec<char> = Vec::<char>::new();
+
+        for l in &self.hits {
+            guessed_vec.push(*l);
+            guessed_vec.push(' ');
+        }
+        guessed_vec.pop();
+
+        let guessed:String = guessed_vec.into_iter().collect();
+
```

Comment:
This space is not necessary, please remove it.

Reply:
Removed